src/Compression/CompressionCodecT64.cpp

```diff
@@ -712,7 +715,7 @@ CompressionCodecT64::CompressionCodecT64(TypeIndex type_idx_, Variant variant_)
 void CompressionCodecT64::updateHash(SipHash & hash) const
 {
     getCodecDesc()->updateTreeHash(hash);
-    hash.update(type_idx);
+    hash.update(type_idx ? *type_idx : TypeIndex::Nothing);
```

Comment:
nit: `type_idx.value_or(TypeIndex::Nothing)` is shorter.

Reply:
Agree, changed